app/helpers/foreman_leapp/template_helper.rb

```diff
@@ -6,17 +6,17 @@ def build_remediation_plan(remediation_ids, host)
       entries = PreupgradeReportEntry.where(id: remediation_ids, host: host)
                                      .where.not(detail: nil)
                                      .pluck(:detail)
-      result = +''
+      commands = []
 
       entries.each do |entry|
         entry['remediations']&.each do |remediation|
           next unless remediation['type'] == 'command'
 
-          result << "#{remediation['context'].join(' ')}\n"
+          commands << "#{remediation['context'].join(' ')}\n"
         end
       end
```

Comment:
Not necessary, but since you're already touching this... How about rewriting this as a series of transformations by chaining various `map`s, `select`s with an occasional `flatten`?